src/atlas.nim

```diff
@@ -457,6 +458,57 @@ proc listOutdated(c: var AtlasContext) =
     listOutdated c, c.depsDir
   listOutdated c, c.workspace
 
+proc newProject(c: var AtlasContext; projectName: string) =
+  ## Tries to create a new project directory in the current dir
+  ## with a single bare `projectname.nim` file inside.
+  ## `projectName` is validated.
+
+  proc isValidProjectName(n: openArray[char]): bool =
+    ## Validates `n` as a project name:
+    ## Valid Nim identifier with addition of dashes (`-`) being allowed,
+    ## but replaced with underscores (`_`) for the `.nim` file name.
+    ## .. Note: Doesn't check if `n` is a valid file/directory name.
+    if not (n.len > 0 and n[0] in IdentStartChars):
+      return false
+    var prevDashOrUnderscore = false
+    for c in n:
+      if c in {'-', '_'}:
+        if prevDashOrUnderscore:
+          return false
+        else:
+          prevDashOrUnderscore = true
+      elif c notin (Letters + Digits):
+        return false
+      else:
+        prevDashOrUnderscore = false
+    true
+
+  let name = block:
+    var n = projectName.strip()
+    if n.isValidFilename() and isValidProjectName(n):
+      n
+    else:
+      error c, toRepo(n), "'" & n & "' is not a vaild project name!"
```

Comment:
Watch out! `error` might delay the error message and not print it!

Reply:
Not sure what to use there. Just outputting to stderr before each quit?

Reply:
Never mind, somebody removed that logic already. :-(